netket/graph/hypercube.py

```diff
@@ -0,0 +1,54 @@
+from custom_graph import PyCustomGraph
+
+import numpy as _np
+import networkx as _nx
+
+class PyHypercube(PyCustomGraph):
```

Comment:
Since through `networkx.generators.lattice.grid_graph` this class now supports non-square grids by passing, e.g., `length=(10, 5)`, maybe it should be renamed (`Hypergrid` or just `Grid`?).

Reply:
Yes, I would call it Grid but I would also leave at least a conveniency function creating Hypercube from one of these more general cases! Also, it might good to have small functions for the most used cases or for people like me who don't remember specific names without looking at the documentation (e.g.):

```python
def Square(L,pbc=True):
    return Grid(length=(L,L),pbc=pbc)
```
or something like that... maybe also for ```Cube``` and ```Line```.

Reply:
I fully agree. A couple of creation functions like `Hypercube`,  `Line`, and `Square` like in your example would be convenient.

Reply:
Shall I make these separate classes that inherit the Grid class properties?